### PR TITLE
sign_up info

### PR DESCRIPTION
**Pull request by chan227**

### CODE REVIEW

**Comment by chan227** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    length: { maximum: Settings.validations.email.max_length },
+    format: { with: Settings.validations.email.regex },
+    uniqueness: true
+
+  validates :password, presence: true,
+    length: { minimum: Settings.validations.password.min_length }
+
+  has_secure_password
+
+  before_save { email.downcase! }
```
Comment (ngocvt-0484) [inline app/models/user.rb]: please write method for email.downcase!  and call it here like sample_app

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  validates :name, presence: true,
+    length: { maximum: Settings.validations.name.max_length }
+
+  validates :email, presence: true,
+    length: { maximum: Settings.validations.email.max_length },
+    format: { with: Settings.validations.email.regex },
+    uniqueness: true
+
+  validates :password, presence: true,
+    length: { minimum: Settings.validations.password.min_length }
```
Comment (ngocvt-0484) [inline app/models/user.rb]: missing allow_nil: true

Comment (chan227) [reply app/models/user.rb]: done

File: app/models/user.rb (hunk 1 above)
Comment (chan227) [reply app/models/user.rb]: done

**Comment by chan227** [comment]
ready

**Review by saobv-2223 — COMMENTED** [review]
merge

**Pull request merged**